scribble.go

```diff
@@ -199,6 +199,39 @@ func (d *Driver) ReadAll(collection string) ([]string, error) {
 	return records, nil
 }
 
+// List all records from a collection; this is returned as a slice of strings because
+// there is no way of knowing what type the record is.
+func (d *Driver) List(collection string) ([]string, error) {
+
+	// ensure there is a collection to read
+	if collection == "" {
+		return nil, fmt.Errorf("missing collection - unable to record location")
+	}
+
+	//
+	dir := filepath.Join(d.dir, collection)
+
+	// check to see if collection (directory) exists
+	if _, err := stat(dir); err != nil {
+		return nil, nil
+	}
+
+	// read all the files in the transaction.Collection; an error here just means
+	// the collection is either empty or doesn't exist
+	files, _ := ioutil.ReadDir(dir)
+
+	// the IDs of collection
+	var records []string
+
+	for _, file := range files {
+		name := file.Name()
+		records = append(records, name[:len(name)-5])
```

Comment:
Index out of range? Or at the very least an empty string? Not sure why we want to truncate 5 characters.

Reply:
"Index out of range" - that's a good point.
I assumed that in the database directory there are only .json files, but in fact someone could put some other file there.

Fix for that:
```
	for _, file := range files {
		name := file.Name()
		if strings.HasSuffix(name, ".json") {
			records = append(records, name[:len(name)-5])
		}
	}
```

"Not sure why we want to truncate 5 characters."
The proposed List() function returns the ID of the objects from the selected collection.
To return the ID list, I list all files in the collection directory with `files, _ := ioutil.ReadDir(dir)`. Then, from the file names, which in practice are the object IDs, remove .json from the end of the file name.
Similarly to what you did in the ReadAll() function, only that in List() I do not open object files, I am only interested in the list of their name / ID. With the object ID, I can use the existing Read() function to read the object.

What solves the problem that is contained in the description of the ReadAll() function.
"because there is no way of knowing what type the record is."